### ISSUE TITLE
Testing breaks when site has a catch-all route

### ISSUE THREAD

**Issue by nielspeen** [livewire/livewire#748]
TestableLivewire.php registers a new Route for every test. The route gets placed at the bottom of the routing table.

This will fail on sites that have a catch-all route. (E.g. to handle slugs, or provide some kind of intelligent feedback/redirection for invalid URLs.)

It's easy to work-around by filtering the catch-all (see below), but I think this should not be necessary.

```
Route::->get('{url}', 'CatchAllController@show')
             ->where('url', '^(?!(testing-livewire\/).*$');
```

**Comment by basepack** [comment]
Same here:

**web.php**
```
    Route::get( '/{category}', [ CategoryController::class, 'index' ] )->name( 'categories.index' );
    Route::get( '/{category}/{subCategory?}', [ CategoryController::class, 'index' ] )->name( 'categories.index' );

```
**CategoryController.php**
```
use App\Http\Queries\ProductQuery;

class CategoryController extends Controller
{

    public function index( ProductQuery $productQuery, Category $category, ?Category $subCategory = null )
    {
       ...
    }
}
```

**Comment by basepack** [comment]
Is it possible to designate a specific route for testing Livewire Components?

**Comment by sheldonkotyk** [comment]
I think we're running into this with Statamic 3 as it takes over routing if something isn't set. :/

**Comment by DanHarrin** [comment]
Hi @sheldonkotyk and @basepack, #980 was merged last month which should fix this issue. If these issues are still present, are you able to provide a set of replication instructions for them please? Many thanks.

**Comment by calebporzio** [comment]
Re-submit if it hasn't been fixed. Thanks!

**Comment by basepack** [comment]
Fixed with #980, thanks!
